homework/grayscale-image/compression.cpp

```diff
@@ -0,0 +1,58 @@
+#include "compression.hpp"
+#include <algorithm>
+#include <iostream>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& image) {
+	std::vector<std::pair<uint8_t, uint8_t>> compressedImage;
+	compressedImage.reserve(width * height);
+	for (auto verticalIter = image.cbegin(); verticalIter != image.cend(); verticalIter++) {
+		std::vector<uint8_t> elements;
+		elements.reserve(width);
+		std::vector<uint8_t> amounts(width);
+		std::fill(amounts.begin(), amounts.end(), 1);

```

Comment:
you can do amounts(width, 1) is more efficient. Why you initilaize everything with 1?